components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -436,88 +334,30 @@ public Response getAllApplicationsByUser(@PathParam("userId") String userId,@Pat
 
 	@GET
 	@Path("/apis/{userId}/{appId}")
-	@Produces("application/json")
-	public Response getAllApisByUserAndApp(@PathParam("userId") String userId,@PathParam("appId") String appId) throws SQLException, BusinessException {
-		//Gson gson = new GsonBuilder().serializeNulls().create();
-		if (userId!=null && appId!=null) {
-			String jsonString = blackListWhiteListService.getAllApisByUserAndApp(userId,appId);
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response getAllApisByUserAndApp(@PathParam("userId") String userId, @PathParam("appId") String appId) {
+		String jsonString = null;
+		try {
+			userId = replaceAllWithWildcard(userId);
+			appId = replaceAllWithWildcard(appId);
+
+			jsonString = blackListWhiteListService.getAllApisByUserAndApp(userId, appId);
 			return Response.status(Response.Status.OK).entity(jsonString).build();
+		} catch (BusinessException e) {
+			return Response.status(Response.Status.BAD_REQUEST).build();
 		}
-		return Response.status(Response.Status.BAD_REQUEST).build();
+
+
 	}
 
 	@GET
 	@Path("/apis")
-	@Produces("application/json")
+	@Produces(MediaType.APPLICATION_JSON)
 	public Response getAllApi() throws SQLException, BusinessException {
```

Comment:
will remove the SQLException which is not thrown from the method

Reply:
updated